frontend/packages/console-shared/src/components/health-checks/HealthChecksAlert.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Link } from 'react-router-dom';
+import { Alert, AlertActionCloseButton } from '@patternfly/react-core';
+import { ServiceModel as KnativeServiceModel } from '@console/knative-plugin';
+import { K8sResourceKind, K8sKind, referenceForModel } from '@console/internal/module/k8s';
+import {
+  DeploymentConfigModel,
+  DeploymentModel,
+  DaemonSetModel,
+  StatefulSetModel,
+} from '@console/internal/models';
+import { STORAGE_PREFIX } from '../../constants';
+import './HealthChecksAlert.scss';
+
+type HealthChecksAlertProps = {
+  resource: K8sResourceKind;
+  model: K8sKind;
+};
+
+const HIDE_HEALTH_CHECK_ALERT_FOR = `${STORAGE_PREFIX}/hide-health-check-alert-for`;
+
+const addHealthChecksRefs = [
+  referenceForModel(DeploymentConfigModel),
+  referenceForModel(DeploymentModel),
+  referenceForModel(DaemonSetModel),
+  referenceForModel(StatefulSetModel),
+  referenceForModel(KnativeServiceModel),
+];
+
+const HealthChecksAlert: React.FC<HealthChecksAlertProps> = ({ resource, model }) => {
+  const [hideHealthCheckAlertFor, setHideHealthCheckAlertFor] = React.useState([]);
+
+  React.useEffect(() => {
+    setHideHealthCheckAlertFor(JSON.parse(localStorage.getItem(HIDE_HEALTH_CHECK_ALERT_FOR)) || []);
+  }, []);
+
+  if (!_.includes(addHealthChecksRefs, referenceForModel(model))) {
+    return null;
+  }
+
+  const containers = resource?.spec?.template?.spec?.containers;
+  const containersName = containers?.map((container) => container.name);
+  const healthCheckAdded = _.every(
+    containers,
+    (container) => container.readinessProbe || container.livenessProbe || container.startupProbe,
+  );
+
+  const handleAlertAction = (name: string, namespace: string) => {
+    const hideHealthCheckAlert = [...hideHealthCheckAlertFor, `${namespace}/${name}`];
```

Comment:
`${namespace}/${name}` is not a unique identifier because I can have a two different workload types which have the same name.
You are better off using the UID instead so that if someone deletes a workload and then recreates it with the same name, they are prompted again.